packages/gatsby/src/utils/page-data.js

```diff
@@ -2,10 +2,14 @@ const fs = require(`fs-extra`)
 const path = require(`path`)
 const { store } = require(`../redux`)
 
-const getFilePath = ({ publicDir }, pagePath) => {
+const getFilePath = ({ publicDir }, pagePath, type) => {
   const fixedPagePath = pagePath === `/` ? `index` : pagePath
+  if (type === `html`) {
+    return path.join(publicDir, fixedPagePath, `index.html`)
```

Comment:
I think this will be wrong for `/` path:

it would get `fixedPagePath` equal to `index`

and entire return would be `public/index/index.html` (it should be `public/index.html`)

I also don't think we should overload this function to handle both `page-data` and `html` "types" to be honest.

We currently have this function that turns page path into `html` file path:
https://github.com/gatsbyjs/gatsby/blob/80acb8d5d67f7e277ce44158b36da84d262e5b23/packages/gatsby/src/utils/worker/render-html.js#L5-L14

It's not exported anywhere right now, but I think we should reuse that to make sure we only have single implementation of this function so we never desync:

I would propose creating `utils/page-html.js` utility module and moving that function there. Import it from `render-html.js` file I linked above and also import it here (at least for now - I still don't like that we handle .html files in `page-data` utils module - but it's for sure better than what we had initially when we were constructing file paths in `commands/build.js` - will think if we can come up with some nicer organization of files/modules so concerns are not leaking from the modules etc)

Reply:
🤦‍♂ I'm making silly mistakes here, thank you for picking up that. 
I've reused that function and moved it to a new `utils/page-html.js ` I've also made a slight change in the name and params.